BasicSample/app/src/main/java/com/example/android/persistence/ui/CommentAdapter.java

```diff
@@ -41,41 +43,53 @@ public CommentAdapter(@Nullable CommentClickCallback commentClickCallback) {
-                @Override
-                public int getOldListSize() {
-                    return mCommentList.size();
-                }
-
-                @Override
-                public int getNewListSize() {
-                    return comments.size();
-                }
+            mCommentList = comments;
+            new AsyncTask<Void, Void, DiffUtil.DiffResult>() {
 
                 @Override
-                public boolean areItemsTheSame(int oldItemPosition, int newItemPosition) {
-                    Comment old = mCommentList.get(oldItemPosition);
-                    Comment comment = comments.get(newItemPosition);
-                    return old.getId() == comment.getId();
+                protected DiffUtil.DiffResult doInBackground(Void... voids) {
+                    return DiffUtil.calculateDiff(new DiffUtil.Callback() {
+                        @Override
+                        public int getOldListSize() {
+                            return mCommentList.size();
+                        }
+
+                        @Override
+                        public int getNewListSize() {
+                            return comments.size();
+                        }
+
+                        @Override
+                        public boolean areItemsTheSame(int oldItemPosition, int newItemPosition) {
+                            Comment old = mCommentList.get(oldItemPosition);
+                            Comment comment = comments.get(newItemPosition);
+                            return old.getId() == comment.getId();
+                        }
+
+                        @Override
+                        public boolean areContentsTheSame(int oldItemPosition, int newItemPosition) {
+                            Comment old = mCommentList.get(oldItemPosition);
+                            Comment comment = comments.get(newItemPosition);
+                            return old.getId() == comment.getId()
+                                    && old.getPostedAt() == comment.getPostedAt()
+                                    && old.getProductId() == comment.getProductId()
+                                    && Objects.equals(old.getText(), comment.getText());
+                        }
+                    });
                 }
 
                 @Override
-                public boolean areContentsTheSame(int oldItemPosition, int newItemPosition) {
-                    Comment old = mCommentList.get(oldItemPosition);
-                    Comment comment = comments.get(newItemPosition);
-                    return old.getId() == comment.getId()
-                            && old.getPostedAt() == comment.getPostedAt()
-                            && old.getProductId() == comment.getProductId()
-                            && Objects.equals(old.getText(), comment.getText());
+                protected void onPostExecute(DiffUtil.DiffResult diffResult) {
+                    super.onPostExecute(diffResult);
```

Comment:
I don't see the purpose or benefit of calling super.onPostExecute(diffResult) here. What am i missing?